catalog/nsili/catalog-nsili-common/src/main/java/org/codice/alliance/nsili/common/ResultDAGConverter.java

```diff
@@ -336,27 +430,42 @@ public static void addCoverageNodeWithAttributes(DirectedAcyclicGraph<Node, Edge
     public static void addThumbnailRelatedFile(DirectedAcyclicGraph<Node, Edge> graph,
             Node productNode, Metacard metacard, ORB orb) {
         Any any = orb.create_any();
-        Node relatedFileNode = new Node(0, NodeType.ENTITY_NODE, NsiliConstants.NSIL_RELATED_FILE, any);
+        Node relatedFileNode = new Node(0,
+                NodeType.ENTITY_NODE,
+                NsiliConstants.NSIL_RELATED_FILE,
+                any);
         graph.addVertex(relatedFileNode);
         graph.addEdge(productNode, relatedFileNode);
 
         Attribute pocAttr = metacard.getAttribute(Metacard.POINT_OF_CONTACT);
         if (pocAttr != null) {
             String pocString = String.valueOf(pocAttr.getValue());
-            if (pocString != null) {
+            if (StringUtils.isNotBlank(pocString)) {
                 addStringAttribute(graph, relatedFileNode, NsiliConstants.CREATOR, pocString, orb);
+            } else {
+                addStringAttribute(graph, relatedFileNode, NsiliConstants.CREATOR, SystemInfo.getSiteName(), orb);
             }
+        } else {
+            addStringAttribute(graph, relatedFileNode, NsiliConstants.CREATOR, SystemInfo.getSiteName(), orb);
         }
 
-        addDateAttribute(graph,
-                relatedFileNode,
-                NsiliConstants.DATE_TIME_DECLARED,
-                metacard.getCreatedDate(),
-                orb);
+        if (metacard.getCreatedDate() != null) {
+            addDateAttribute(graph,
+                    relatedFileNode,
+                    NsiliConstants.DATE_TIME_DECLARED,
+                    metacard.getCreatedDate(),
+                    orb);
+        } else {
+            addDateAttribute(graph,
+                    relatedFileNode,
+                    NsiliConstants.DATE_TIME_DECLARED,
+                    new Date(),
+                    orb);
+        }
 
-        if (metacard.getResourceSize() != null) {
+        if (metacard.getThumbnail() != null) {
```

Comment:
The RelatedFile is how you reference a Thumbnail in Nsili. It supports an EXTENT (size). This is what is being filled in.